vignettes/advs.Rmd

```diff
@@ -0,0 +1,392 @@
+)
+attr(param_lookup$VSTESTCD, "label") <- "Vital Signs Test Short Name"
+```
+
+This lookup may now be joined to the source data and this is how the parameters will look like.
+```{r, eval=TRUE, include=FALSE}
+advs <- derive_vars_merged_lookup(
+  advs,
+  dataset_add = param_lookup,
+  new_vars = exprs(PARAMCD, PARAM, PARCAT1, PARCAT1N),
+  by_vars = exprs(VSTESTCD)
+)
+
+advs <- mutate(
+  advs,
+  AVAL = VSSTRESN
+)
+```
+
+```{r, eval=TRUE, echo=FALSE}
+advs_param <- distinct(advs, USUBJID, PARAMCD, VSTESTCD, PARAM, PARCAT1, PARCAT1N)
+
+dataset_vignette(advs_param, display_vars = exprs(USUBJID, VSTESTCD, PARAMCD, PARAM, PARCAT1, PARCAT1N))
+```
+
+## Derive Additional Parameters (e.g. `BMI` for `ADVS`) {#derive_param}
+
+In metabolic trials, `BMI` is often calculated at source using collected
+height and weight for screening purposes. But while creating the `ADVS`
+dataset, we re-derive `BMI` from the collected height and weight values.
+This is done to ensure that the `BMI` is calculated consistently across
+all subjects and visits.
+
+In this step, we create parameter `Body Mass Index` (BMI) for `ADVS`
+domain using the `derive_param_bmi()` function. Note that only variables
+specified in the `by_vars` argument will be populated in the newly
+created records. "Also note that if height is collected only once for a
+subject use `constant_by_vars` to specify the function to merge by the
+subject-level variable" Otherwise BMI is only calculated for visits
+where both are collected.
+
+```{r eval=TRUE}
+# Removing BMI collected at source from the dataset
+advs <- advs %>% filter(!VSTESTCD == "BMI")
+
+advs <- derive_param_bmi(
+  advs,
+  by_vars = c(
+    get_admiral_option("subject_keys"),
+    exprs(!!!adsl_vars, VISIT, VISITNUM, ADT, ADY, VSTPT, VSTPTNUM)
+  ),
+  set_values_to = exprs(
+    PARAMCD = "BMI",
+    PARAM = "Body Mass Index (kg/m^2)",
+    PARAMN = 3,
+    PARCAT1 = "Subject Characteristic",
+    PARCAT1N = 1
+  ),
+  get_unit_expr = VSSTRESU,
+  filter = VSSTAT != "NOT DONE" | is.na(VSSTAT),
+  constant_by_vars = exprs(USUBJID)
+)
+```
+
+```{r, eval=TRUE, echo=FALSE}
+dataset_vignette(
+  arrange(advs, USUBJID, VISITNUM, VSTPTNUM, ADT, PARAMCD),
+  display_vars = exprs(USUBJID, VSTESTCD, PARAMCD, PARAM, VISIT, AVAL),
+  filter = PARAMCD %in% c("BMI")
+)
+```
+
+```{r eval=TRUE, include=FALSE}
+advs <- restrict_derivation(
+  advs,
+  derivation = derive_var_extreme_flag,
+  args = params(
+    by_vars = c(get_admiral_option("subject_keys"), exprs(PARAMCD)),
+    order = exprs(ADT, VSTPTNUM, VISITNUM),
+    new_var = ABLFL,
+    mode = "last"
+  ),
+  filter = (!is.na(AVAL) & ADT <= TRTSDT)
+)
+
+advs <- derive_var_base(
+  advs,
+  by_vars = c(get_admiral_option("subject_keys"), exprs(PARAMCD)),
+  source_var = AVAL,
+  new_var = BASE
+)
+
+advs <- derive_var_chg(advs)
+
+advs <- derive_var_pchg(advs)
+```
+
+In the following sections, we will explore some of the most common
```

Comment:
I think this would read better if you add a new section header here (maybe "Common Metabolism Endpoints") and make the next two sections into subsections.